tests/test_subgridpart.cpp

```diff
@@ -0,0 +1,250 @@
+            continue;
+        }
+        testElement(gridView, *elemIt);
+        ++ numElem;
+    }
+
+    BOOST_CHECK_EQUAL(numElem, nElem);
+}
+
+template <class GridView, Dune::PartitionIteratorType pit>
+void testGridPartitionIteration( const GridView& gridView, const int nElem)
+{
+    int numElem = 0;
+    auto elemIt = gridView.template begin<0, pit>();
+    const auto elemEndIt = gridView.template end<0, pit>();
+    for (; elemIt != elemEndIt; ++elemIt) {
+        testElement(gridView, *elemIt);
+        ++ numElem;
+    }
+
+    BOOST_CHECK_EQUAL(numElem, nElem);
+}
+
+
+template <class Grid>
+auto getSeeds(const Grid& grid, const std::vector<int>& indices)
+{
+    assert(std::is_sorted(indices.begin(), indices.end()));
+    using EntitySeed = typename Grid::template Codim<0>::Entity::EntitySeed;
+    std::vector<EntitySeed> seeds(indices.size());
+    auto it = grid.template leafbegin<0>();
+    int previous = 0;
+    for (std::size_t c = 0; c < indices.size(); ++c) {
+        std::advance(it, indices[c] - previous);
+        seeds[c] = it->seed();
+        previous = indices[c];
+    }
+    return seeds;
+}
+
+
+template <class Grid>
+void testGrid(Grid& grid, const std::string& name, const std::size_t nElem, const std::size_t nVertices)
+{
+    typedef typename Grid::LeafGridView GridView;
+
+    std::cout << name << std::endl;
+
+    testGridInteriorIteration( grid.leafGridView(), nElem );
+
+    std::cout << "create vertex mapper\n";
+    Dune::MultipleCodimMultipleGeomTypeMapper<GridView> mapper(grid.leafGridView(), Dune::mcmgVertexLayout());
+
+    BOOST_CHECK_EQUAL(mapper.size(), nVertices);
+
+    {
+        // Testing with overlap
+        Dune::SubGridPart<Grid> sgv(grid, getSeeds(grid, {0, 1, 2}), true);
+        const std::size_t sz = sgv.size(0);
+        BOOST_CHECK(sz > 3);
+        const std::size_t overlap = sgv.overlapSize(0);
+        BOOST_CHECK_EQUAL(sz, overlap + 3);
+        testGridInteriorIteration(sgv, 3);
+        testGridPartitionIteration<Dune::SubGridPart<Grid>, Dune::Interior_Partition>(sgv, 3);
+        testGridPartitionIteration<Dune::SubGridPart<Grid>, Dune::All_Partition>(sgv, sz);
+        testGridPartitionIteration<Dune::SubGridPart<Grid>, Dune::Overlap_Partition>(sgv, overlap);
+    }
+    {
+        // Testing without overlap
+        Dune::SubGridPart<Grid> sgv(grid, getSeeds(grid, {0, 1, 2}), false);
+        testGridInteriorIteration(sgv, 3);
+        testGridPartitionIteration<Dune::SubGridPart<Grid>, Dune::Interior_Partition>(sgv, 3);
+        testGridPartitionIteration<Dune::SubGridPart<Grid>, Dune::All_Partition>(sgv, 3);
+        testGridPartitionIteration<Dune::SubGridPart<Grid>, Dune::Overlap_Partition>(sgv, 0);
+    }
+}
+
+
+#if HAVE_ECL_INPUT
+BOOST_AUTO_TEST_CASE(FromDeck)
+{
+    // ------------ Test grid from deck. ------------
+    const char* deckString =
+R"(
+RUNSPEC
+METRIC
+DIMENS
+2 2 2 /
+GRID
+DXV
+2*1 /
+DYV
+2*1 /
+DZ
+8*1 /
+TOPS
+8*100.0 /
```

Comment:
I probably doesn't matter a whole lot, but this `8 * 100` should **probably** be `4 * 100`, possibly followed by `4 * 101`.  As written, this `TOPS` vector does not represent a valid tesselation of the geometry.

Reply:
Done.